lib/client.js

```diff
@@ -67,8 +67,10 @@ class FlowManagerClient {
         this._setConsumerStatus({ healthy: false, ready: false })
         this.log.info('consumer disconnected')
       }),
-      this.consumer.on(CRASH, () => {
-        this._setConsumerStatus({ healthy: true, ready: false })
+      this.consumer.on(CRASH, (event) => {
+        // If event.payload.restart === true the consumer will try to restart itself.
+        // If the error is not retriable, the consumer will instead stop and exit.
+        this._setConsumerStatus({ healthy: event.payload.restart, ready: false })
```

Comment:
This event is generated from kafkajs. It is set to true if the consumer will try to restart on its own, false otherwise.
The idea is to set the health: `false` if the consumer is dead in a way that it will not retry to connect to the brokers.